src/Aspire.Hosting/AspireStore.cs

```diff
@@ -0,0 +1,234 @@
+
+        if (string.IsNullOrEmpty(root))
+        {
+            throw new InvalidOperationException($"Could not determine an appropriate location for storing user secrets. Set the {AspireStorePathKeyName} setting to a folder where the App Host content should be stored.");
+        }
+
+        var directoryPath = Path.Combine(root, ".aspire");
+
+        // The /obj directory doesn't need to be prefixed with the app host name.
+        if (root != objDir)
+        {
+            directoryPath = Path.Combine(directoryPath, GetAppHostSpecificPrefix(builder));
+        }
+
+        return new AspireStore(directoryPath);
+    }
+
+    /// <summary>
+    /// Gets the metadata value for the specified key from the assembly metadata.
+    /// </summary>
+    /// <param name="assemblyMetadata">The assembly metadata.</param>
+    /// <param name="key">The key to look for.</param>
+    /// <returns>The metadata value if found; otherwise, null.</returns>
+    private static string? GetMetadataValue(IEnumerable<AssemblyMetadataAttribute>? assemblyMetadata, string key) =>
+        assemblyMetadata?.FirstOrDefault(a => string.Equals(a.Key, key, StringComparison.OrdinalIgnoreCase))?.Value;
+
+    /// <summary>
+    /// Gets the application host specific prefix based on the builder's environment.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <returns>The application host specific prefix.</returns>
+    private static string GetAppHostSpecificPrefix(IDistributedApplicationBuilder builder)
+    {
+        var appName = Sanitize(builder.Environment.ApplicationName).ToLowerInvariant();
+        var appNameHash = builder.Configuration["AppHost:Sha256"]![..10].ToLowerInvariant();
+        return $"{appName}.{appNameHash}";
+    }
+
+    /// <summary>
+    /// Gets a deterministic file path that is a copy of the <paramref name="sourceFilename"/>.
+    /// The resulting file name will depend on the content of the file.
+    /// </summary>
+    /// <param name="filename">A file name to base the result on.</param>
+    /// <param name="sourceFilename">An existing file.</param>
+    /// <returns>A deterministic file path with the same content as <paramref name="sourceFilename"/>.</returns>
+    /// <exception cref="FileNotFoundException">Thrown when the source file does not exist.</exception>
+    public string GetFileNameWithContent(string filename, string sourceFilename)
+    {
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(filename);
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(sourceFilename);
+
+        if (!File.Exists(sourceFilename))
+        {
+            throw new FileNotFoundException("The source file '{0}' does not exist.", sourceFilename);
+        }
+
+        EnsureDirectory();
+
+        // Strip any folder information from the filename.
+        filename = Path.GetFileName(filename);
+
+        // Delete existing file versions with the same name.
```

Comment:
I'm not sure we want to do this. What if these files are being used?

Reply:
The point of putting this in "/obj" is that we can leave these orphaned files, and when the user deletes the whole project or cleans their repo, the files will go away.

Reply:
The point was that there is still the fallback folder that is handled, and it's a cheap effort to clean older versions of a file we are about to update. If you think there is some risk associated with it I'll remove it.

Reply:
I just don't think we should be messing with any other file than what we are operating on. We don't know who else is using these files. Someone could call this API twice with different contents and actually want 2 different files.

Callers and users who want things cleaned up can do it outside of this core API.